### PR TITLE
prepareObject() implementation for DoctrineOrmMapper

### PR DESCRIPTION
**Pull request by bitladen-nw**
while implementing collections using ORM i came accross the missing implementation for `DoctrineOrmMapper:prepareObject()`... well, here is mine in collaboration with dbu.

### CODE REVIEW

File: src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php [hunk 1]
```diff
@@ -37,6 +40,54 @@ class DoctrineOrmMapper extends BaseDoctrineRdfMapper
...
+
+
+    /**
+     * find the parent object's property which holds the collection entries
+     *
+     * @param $parent
+     * @param ClassMetadata $metaData metadata from the collection entry's entity
+     * @return bool|string
+     */
+    protected function findParentMapping($parent, ClassMetadata $metaData = null)
```
Comment (dbu) [inline src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php]: i would also pass the object itself to this method for convenience.

File: src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php [hunk 2]
```diff
@@ -37,6 +40,54 @@ class DoctrineOrmMapper extends BaseDoctrineRdfMapper
...
+    {
+        $parentClass = ClassUtils::getRealClass(get_class($parent));
+
+        foreach ($metaData->associationMappings as $mapping) {
+            if ($mapping['targetEntity'] == $parentClass) {
+                return $mapping['fieldName'];
+            }
+        }
+
+        return false;
```
Comment (dbu) [inline src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php]: i suggest that we throw an exception instead, something like "No mapping for parent class $parentClass found in metadata of $metaData->getName"

File: src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php (hunk 1 above)
Comment (dbu) [inline src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php]: and metaData is not allowed to be null

File: src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php [hunk 3]
```diff
@@ -37,6 +40,54 @@ class DoctrineOrmMapper extends BaseDoctrineRdfMapper
...
+                . get_class($object));
+    }
+
+
+    /**
+     * find the parent object's property which holds the collection entries
+     *
+     * @param $parent
+     * @param ClassMetadata $metaData metadata from the collection entry's entity
+     * @return bool|string
```
Comment (dbu) [inline src/Midgard/CreatePHP/Mapper/DoctrineOrmMapper.php]: if we throw the exception, this will never be bool but always a string.

**Comment by dbu** [comment]
@flack this looks ok from my point of view. any objections if i squash the commits and then merge it?

**Comment by flack** [comment]
@dbu Squashing seems like a good idea, and the code looks fine, too (admittedly, I only looked over it briefly, but still :-))

Thanks for the contribution, @bitladen-nw !

**Comment by dbu** [comment]
continued in #75

**Pull request closed without merging**